x-pack/packages/kbn-elastic-assistant/impl/assistant/use_conversation/index.test.tsx

```diff
@@ -285,91 +186,58 @@ describe('useConversation', () => {
+        conversationId: welcomeConvo.id,
+        replacements: {
+          '1.0.0.721': '127.0.0.1',
+          '1.0.0.01': '10.0.0.1',
+          'tsoh-tset': 'test-host',
         },
       });
     });
   });
 
   it('should remove the last message from a conversation when called with valid conversationId', async () => {
     await act(async () => {
-      const setConversations = jest.fn();
       const { result, waitForNextUpdate } = renderHook(() => useConversation(), {
         wrapper: ({ children }) => (
-          <TestProviders
-            providerContext={{
-              getInitialConversations: () => ({
-                [alertConvo.id]: alertConvo,
-                [welcomeConvo.id]: welcomeConvo,
-                [mockConvo.id]: mockConvo,
-              }),
-              setConversations,
-            }}
-          >
-            {children}
-          </TestProviders>
+          <TestProviders providerContext={{ http: httpMock }}>{children}</TestProviders>
         ),
       });
       await waitForNextUpdate();
 
-      const removeResult = result.current.removeLastMessage('new-convo');
+      getConversationById.mockResolvedValue(mockConvo);
+
+      const removeResult = await result.current.removeLastMessage('new-convo');
 
       expect(removeResult).toEqual([message]);
-      expect(setConversations).toHaveBeenCalledWith({
-        [alertConvo.id]: alertConvo,
-        [welcomeConvo.id]: welcomeConvo,
-        [mockConvo.id]: { ...mockConvo, messages: [message] },
-      });
     });
   });
 
   it('amendMessage updates the last message of conversation[] for a given conversationId with provided content', async () => {
     await act(async () => {
-      const setConversations = jest.fn();
       const { result, waitForNextUpdate } = renderHook(() => useConversation(), {
         wrapper: ({ children }) => (
-          <TestProviders
-            providerContext={{
-              getInitialConversations: () => ({
-                [alertConvo.id]: alertConvo,
-                [welcomeConvo.id]: welcomeConvo,
-                [mockConvo.id]: mockConvo,
-              }),
-              setConversations,
-            }}
-          >
-            {children}
-          </TestProviders>
+          <TestProviders providerContext={{ http: httpMock }}>{children}</TestProviders>
         ),
       });
       await waitForNextUpdate();
 
-      result.current.amendMessage({
+      await result.current.amendMessage({
         conversationId: 'new-convo',
         content: 'hello world',
       });
 
-      expect(setConversations).toHaveBeenCalledWith({
-        [alertConvo.id]: alertConvo,
-        [welcomeConvo.id]: welcomeConvo,
-        [mockConvo.id]: {
-          ...mockConvo,
-          messages: [
-            message,
-            {
-              ...anotherMessage,
-              content: 'hello world',
-            },
-          ],
-        },
+      expect(appendConversationMessagesApi).toHaveBeenCalledWith({
```

Comment:
I'm wondering if the newly-asserted behavior in this test is correct (desk testing it is currently blocked by https://github.com/elastic/kibana/pull/173487#issuecomment-1912217142), because the previous implementation appears to update the last message, but the new implementation appears to append a message.